src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js

```diff
@@ -1,68 +1,56 @@
 // @flow
-import React, { useState, useMemo, useCallback } from 'react';
+import React, { useState, useCallback } from 'react';
+import type { ComponentType } from 'react';
 import i18n from '@dhis2/d2-i18n';
+import { Button, spacersNum } from '@dhis2/ui';
+import { withStyles } from '@material-ui/core';
 import log from 'loglevel';
 import { FlatList } from 'capture-ui';
 import { errorCreator } from 'capture-core-utils';
-import { useDataQuery } from '@dhis2/app-runtime';
 import { Widget } from '../Widget';
 import { LoadingMaskElementCenter } from '../LoadingMasks';
 import { convertValue as convertClientToView } from '../../converters/clientToView';
 import { convertValue as convertServerToClient } from '../../converters/serverToClient';
 import type { Props } from './widgetProfile.types';
+import { DataEntry } from './DataEntry';
+import { useProgram, useTrackedEntityInstances } from './hooks';
 
-export const WidgetProfile = ({ teiId, programId }: Props) => {
-    const [open, setOpenStatus] = useState(true);
-    const programsQuery = useMemo(() => ({
-        programs: {
-            resource: 'programs',
-            id: programId,
-            params: {
-                fields:
-                ['programTrackedEntityAttributes[id,displayInList,trackedEntityAttribute[id,displayName,valueType]]'],
-            },
-        },
-    }), [programId]);
-
-    const trackedEntityInstancesQuery = useMemo(() => ({
-        trackedEntityInstances: {
-            resource: 'trackedEntityInstances',
-            id: teiId,
-            params: {
-                program: programId,
-            },
-        },
-    }), [teiId, programId]);
-
+const styles = {
+    header: {
+        display: 'flex',
+        alignItems: 'center',
+        padding: spacersNum.dp8,
+        justifyContent: 'space-between',
+        width: '100%',
+    },
+};
 
-    const {
-        loading: programsLoading,
-        data: programsData,
-        error: programsError,
-    } = useDataQuery(programsQuery);
+const WidgetProfilePlain = ({ teiId, programId, showEdit = false, orgUnitId = '', classes }: Props) => {
+    const [open, setOpenStatus] = useState(true);
+    const [toggleEditModal, setToggleEditModal] = useState(false);
```

Comment:
should this take the `showEdit` as initial state?

Reply:
No, `showEdit` and `toggleEditModal` are different variables and in charge of different functionality. `toggleEditModal` opens and closes the edit profile modal, while `showEdit` enable the edit profile feature. Thanks

Reply:
but if there is no showEdit button then how can it open the Edit modal?

Reply:
As mention above the variables and in charge of different functionality.

- `showEdit === false`
       When there is no `showEdit` button then the feature will not be displayed at all. The value of `toggleEditModal` does not matter in this case. 

- `showEdit === true`
      In this case a conflict will appear if `showEdit` is set as the initial state for `toggleEditModal`. The modal will be opened from the initialization. This is not the desired behavious for this feature. 

Thanks!